toolchain/check/eval.cpp

```diff
@@ -2036,19 +2041,22 @@ auto TryEvalTypedInst<SemIR::WhereExpr>(EvalContext& eval_context,
                                more_info.rewrite_constraints);
             info.other_requirements |= more_info.other_requirements;
           }
-        } else {
-          // TODO: Handle `impls` constraints beyond `.Self impls`.
-          info.other_requirements = true;
+          continue;
         }
-      } else {
-        // TODO: Handle other requirements
+
+        // TODO: Handle `impls` constraints beyond `.Self impls`.
         info.other_requirements = true;
+        continue;
```

Comment:
```suggestion
```

If this goes out, won't the addition of `info.other_requirements` on 2053 handle this?

Note, it may be good to shift the flow a little -- there's a lot of nesting going on, and particularly with the nested `continue` it gets a little difficult to track interactions.